### PR TITLE
CON-4698 Prepare Connect changes during variant regeneration

### PR DESCRIPTION
**Pull request by sbofirov**
This PR is about creating/re-creating variants for an exported product and generation of correct Connect changes about that

### CODE REVIEW

File: Subscribers/Article.php [hunk 1]
```diff
@@ -130,7 +153,7 @@ public function extendBackendArticle(\Enlight_Event_EventArgs $args)
                 );
 
 //                if (\Shopware::VERSION != '__VERSION__' && version_compare(\Shopware::VERSION, '4.2.2', '<')) {
```
Comment (SimonBaeumer) [inline Subscribers/Article.php]: Can we remove the commented code? That would be nice :+1:

File: Tests/Shopware/Connect/Subscribers/ArticleTest.php [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+            ->willReturn($repositoryMock);
+
+        $this->connectExport->expects($this->never())
+            ->method('export');
+
+        $this->subscriber->regenerateChangesForArticle(5);
+    }
+
+
+
```
Comment (SimonBaeumer) [inline Tests/Shopware/Connect/Subscribers/ArticleTest.php]: Empty lines

File: Tests/Unit/Components/StatefulVariantRegeneratorTest.php [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+use ShopwarePlugins\Connect\Tests\PhpUnitTestCasePolyfillTrait;
+
+class StatefulVariantRegeneratorTest extends AbstractConnectUnitTest
+{
+    use PhpUnitTestCasePolyfillTrait;
+
+    /**
+     * @var StatefulVariantRegenerator
+     */
+    private $variantRegenerator;
```
Comment (SimonBaeumer) [inline Tests/Unit/Components/StatefulVariantRegeneratorTest.php]: phpdoc

File: Tests/PhpUnitTestCasePolyfillTrait.php [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+namespace ShopwarePlugins\Connect\Tests;
+
+trait PhpUnitTestCasePolyfillTrait
+{
+    /**
+     * @param mixed $arg
+     * @return mixed
+     */
+    private function captureArg(&$arg)
```
Comment (SimonBaeumer) [inline Tests/PhpUnitTestCasePolyfillTrait.php]: moar description

File: Tests/Unit/Components/StatefulVariantRegeneratorTest.php (hunk 3 above)
Comment (sbofirov) [inline Tests/Unit/Components/StatefulVariantRegeneratorTest.php]: what have to be done here ?

**Pull request merged**